lib/ntlmauth/ntlmauth.cc

```diff
@@ -108,10 +108,19 @@ ntlm_fetch_string(const ntlmhdr *packet, const int32_t packet_size, const strhdr
     int32_t o = le32toh(str->offset);
     // debug("ntlm_fetch_string(plength=%d,l=%d,o=%d)\n",packet_size,l,o);
 
-    if (l < 0 || l > NTLM_MAX_FIELD_LENGTH || o + l > packet_size || o == 0) {
-        debug("ntlm_fetch_string: insane data (pkt-sz: %d, fetch len: %d, offset: %d)\n", packet_size,l,o);
+    if (l < 0 || NTLM_MAX_FIELD_LENGTH < l) {
```

Comment:
IMHO, the original order feels more natural, easier to read/grok:

```suggestion
    if (l < 0 || l > NTLM_MAX_FIELD_LENGTH) {
```

The above "item > limit" order is also more consistent with the last l check below.

Reply:
Conflicting with previous requirements that range validation conditions always use the form   ```A < B < C```